src/InContextSignup.js

```diff
@@ -0,0 +1,44 @@
+import {
+    GetIncontextSignupDismissedAtCall,
+    SetIncontextSignupInitiallyDismissedAtCall,
+    SetIncontextSignupPermanentlyDismissedAtCall
+} from './deviceApiCalls/__generated__/deviceApiCalls.js'
+
+export class InContextSignup {
+    permanentlyDismissed = false
+    initiallyDismissed = false
+
+    /**
+     * @param {import("./DeviceInterface/InterfacePrototype").default} device
+     */
+    constructor (device) {
+        this.device = device
+    }
+
+    async init () {
+        const incontextSignupDismissedAt = await this.device.deviceApi.request(new GetIncontextSignupDismissedAtCall(null))
+        this.permanentlyDismissed = Boolean(incontextSignupDismissedAt.permanentlyDismissedAt)
+        this.initiallyDismissed = Boolean(incontextSignupDismissedAt.initiallyDismissedAt)
```

Comment:
this is the logic that was mixed between the extension transport and the `Settings`, we can just make the call directly here and store the data ourselves without it expanding the scope of the Device or Settings